sql-plugin/src/main/scala/org/apache/spark/sql/rapids/predicates.scala

```diff
@@ -316,6 +316,79 @@ case class GpuEqualNullSafe(left: Expression, right: Expression) extends CudfBin
   }
 }
 
+case class GpuEqualToNoNans(left: Expression, right: Expression) extends CudfBinaryComparison
+    with NullIntolerant {
+  override def symbol: String = "="
+  override def outputTypeOverride: DType = DType.BOOL8
+  override def binaryOp: BinaryOp = BinaryOp.EQUAL
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
+    val result = super.doColumnar(lhs, rhs)
```

Comment:
Does CUDF compare NaNs as equal? I thought it didn't because C++ does not, because it follow the ieee standard.

>    The comparisons EQ, GT, GE, LT, and LE, when either or both operands is NaN returns FALSE.
>
>    The comparison NE, when either or both operands is NaN returns TRUE.

I htink we can deleted all of the overrides for doColumnar and just rely on the default implementations.

Reply:
removed doColumnar overrides in last commit